dm/dm/master/server.go

```diff
@@ -86,7 +87,11 @@ var (
 	registerOnce      sync.Once
 	runBackgroundOnce sync.Once
 
-	checkAndAdjustSourceConfigFunc = checkAndAdjustSourceConfig
+	// the difference of below functions is checkAndAdjustSourceConfigForDMCtlFunc will not AdjustCaseSensitive. It's a
+	// compatibility compromise.
+	// When we need to change the implementation of dmctl to OpenAPI, we should notice the user about this change.
```

Comment:
this upgrade case is so complex, we may add breaking change in doc and release note

see [here](https://github.com/pingcap/tiflow/issues/5255#issuecomment-1108148106)

/cc @sunzhaoyang @WizardXiao